configure.ac

```diff
@@ -1273,8 +1273,8 @@ AS_CASE([$host],
 
 AS_CASE([$ccomptype],
   [msvc],
-    [runtime_asm_objects=${arch}nt.${OBJEXT}],
-  [runtime_asm_objects=${arch}.${OBJEXT}])
+    [runtime_asm_objects="\$(ARCH)nt.${OBJEXT}"],
```

Comment:
This change goes against @shindere's gradually trying to do here - that the `configure` system correctly generates the `Makefile` (i.e. reducing the amount of meta-make-and-shell programming in `configure`).

I think there's a possibly simpler way to do this for ocaml-solo5. This change:
```
sed -i -e '/system=linux_elf/a[$(MAKECONF_BUILD_ARCH)-unkown-none],[has_native_backend=yes; arch=$(MAKECONF_OCAML_BUILD_ARCH)],' configure.ac
```
should do away with the need to override `ARCH` and `NATIVE_COMPILER` in `Makefile.config` (i.e. patch `configure.ac` to recognise the solo5 configuration)